js_modules/dagster-ui/packages/ui-core/src/ui/Sidebar/HierarchicalSidebar.tsx

```diff
@@ -0,0 +1,229 @@
+    indexOfLastSelectedNodeRef.current = indexOfLastSelectedNode;
+
+    React.useLayoutEffect(() => {
+      if (indexOfLastSelectedNode !== -1) {
+        rowVirtualizer.scrollToIndex(indexOfLastSelectedNode, {
+          align: 'center',
+          behavior: 'smooth',
+        });
+      }
+      // Only scroll if the selected node changes
+      // otherwise opening/closing nodes will cause us to scroll again because the index changes
+      // if we toggle a node above the selected node
+    }, [indexOfLastSelectedNode, rowVirtualizer]);
+
+    return (
+      <div style={{height: '100%'}}>
+        {loading ? (
+          <Box flex={{direction: 'column', gap: 9}} padding={12}>
+            <Skeleton $height={21} $width="50%" />
+            <Skeleton $height={21} $width="80%" />
+            <Skeleton $height={21} $width="65%" />
+            <Skeleton $height={21} $width="90%" />
+          </Box>
+        ) : (
+          <Container
+            ref={containerRef}
+            style={{padding: 12, paddingLeft: 0}}
+            onKeyDown={(e) => {
+              let nextIndex = 0;
+              if (e.code === 'ArrowDown' || e.code === 'ArrowUp') {
+                nextIndex = indexOfLastSelectedNodeRef.current + (e.code === 'ArrowDown' ? 1 : -1);
+                indexOfLastSelectedNodeRef.current = nextIndex;
+                e.preventDefault();
+                const nextNodeIdx = (nextIndex + renderedNodes.length) % renderedNodes.length;
+                const nextNode = renderedNodes[nextNodeIdx];
+                if (!nextNode) {
+                  return;
+                }
+                setSelectedPath(nextNode.path);
+                onSelectPath(e, nextNode.path);
+              } else if (e.code === 'ArrowLeft' || e.code === 'ArrowRight') {
+                const open = e.code === 'ArrowRight';
+                const node = renderedNodes[indexOfLastSelectedNode];
+                if (!node || node.type !== 'folder') {
+                  return;
+                }
+                setOpenNodes((nodes) => {
+                  const openNodes = new Set(nodes);
+                  if (open) {
+                    openNodes.add(node.path);
+                  } else {
+                    openNodes.delete(node.path);
+                  }
+                  return openNodes;
+                });
+              }
+            }}
+          >
+            <Inner $totalHeight={totalHeight}>
+              {items.map(({index, key, size, start}) => {
+                const node = renderedNodes[index];
+                invariant(node, 'Sidebar node is required');
+
+                const isSelected = selectedPaths.includes(node.path);
+                const isLastSelected = lastSelectedPath === node.path;
+                const isOpen = openNodes.has(node.path);
+
+                return (
+                  <Row $height={size} $start={start} key={key}>
+                    <div data-index={index} ref={rowVirtualizer.measureElement}>
```

Comment:
Btw, I figured out recently that we don't need this extra inner `div`.

We just need to omit `$height` from the row  and then we can add `data-index={index} ref={rowVirtualizer.measureElement}` directly on the Row element